disnake/mentions.py

```diff
@@ -119,12 +132,14 @@ def from_message(cls, message: Message) -> Self:
         )
 
     def to_dict(self) -> AllowedMentionsPayload:
+        # n.b. this runs nearly every time a message is sent
         parse: List[AllowedMentionTypePayload] = []
         data: AllowedMentionsPayload = {}  # type: ignore
```

Comment:
Bit off-topic for the PR, but the bottom `data["parse"] = parse` could be merged into the dict display: `data: ... = {"parse": parse}`
(or even `data: ... = {"parse": parse := []}`, if pyright can infer the type of the list)